### PR TITLE
Add timeout options and some maintenance stuff.

### PR DESCRIPTION
**Pull request by SpacePossum**
I use the service on places where internet is not always very good (on location at some fashion event for example). In order to make sure (most) of the messages go out I do the sending by using a job queue. It would however be useful to be able to tweak the timeouts as passed by the lib to the curllib it uses to allow for even longer timeouts.

I've added some comments to explain why I changed some other details. Let me know what you think :)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,11 +4,15 @@ php:
     - 5.4
     - 5.5
     - 5.6
-    - hhvm
     - 7.0
+    - 7.1
```
Comment (SpacePossum) [inline .travis.yml]: prep. for upcoming 7.1 release

File: phpunit.xml.dist [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    bootstrap="autoload.php"
+>
+    <testsuites>
+        <testsuite name="php-rest-api">
+            <directory>./tests/</directory>
+        </testsuite>
+    </testsuites>
+    <filter>
+        <whitelist>
+            <directory suffix=".php">./src/</directory>
```
Comment (SpacePossum) [inline phpunit.xml.dist]: allows for PHPUnit to generate a coverage report (and easy PHPUnit runs from the commandline, i.e. just `phpunit`)

File: src/MessageBird/Common/HttpClient.php [hunk 3]
```diff
@@ -27,19 +27,51 @@ class HttpClient
...
+
+    /**
      * @param string $endpoint
+     * @param int $timeout > 0
+     * @param int $connectionTimeout >= 0
+     *
+     * @throws \InvalidArgumentException if timeout settings are invalid
      */
-    public function __construct($endpoint)
+    public function __construct($endpoint, $timeout = 10, $connectionTimeout = 2)
```
Comment (SpacePossum) [inline src/MessageBird/Common/HttpClient.php]: added timeout options without BC breaks

File: src/MessageBird/Common/HttpClient.php [hunk 4]
```diff
@@ -117,10 +150,11 @@ public function performHttpRequest($method, $resourceName, $query = null, $body
         }
 
         // Some servers have outdated or incorrect certificates, Use the included CA-bundle
-        $caFile = realpath(dirname(__FILE__) . "/../ca-bundle.crt");
+        $caFile = realpath(__DIR__ . '/../ca-bundle.crt');
         if (!file_exists($caFile)) {
-            throw new Exceptions\HttpException('Unable to find CA-bundle file: ' . $caFile);
+            throw new Exceptions\HttpException(sprintf('Unable to find CA-bundle file "%s".', __DIR__ . '/../ca-bundle.crt'));
```
Comment (SpacePossum) [inline src/MessageBird/Common/HttpClient.php]: if the file does not exists `realpath` returns `false` so printing the var. `$caFile` did not help

**Comment by SpacePossum** [comment]
btw. fixes https://github.com/messagebird/php-rest-api/issues/19 as well ;)

**Comment by SpacePossum** [comment]
TODO
https://github.com/messagebird/php-rest-api/blob/master/src/MessageBird/Client.php#L139
this check can go as well.
The `constant` was added to PHP  5.2.7 (http://php.net/manual/en/reserved.constants.php) and the project requires PHP 5.3 https://github.com/messagebird/php-rest-api/blob/master/composer.json#L14
so it will always be defined.

**Comment by samwierema** [comment]
Thanks for the pull request @SpacePossum! This will be released as version `1.6.0`.

**Comment by SpacePossum** [comment]
thank you @samwierema :) :+1:

**Pull request merged**